implementations/generate.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Generates the HTTP Signatures Implementation Report given
+ * a set of *-report.json files.
+ */
+const fs = require('fs');
+const path = require('path');
+const testConfig = require('./test.json');
+
+// extract the results from all of the test files
+const allTests = [];
+const allResults = {};
+const dirContents = fs.readdirSync(__dirname);
+const files = dirContents.filter(
+  contents => {return contents.match(/.*-report.json/ig);});
+
+if(!files.length) {
+  throw new Error(
+    'Failed to find any reports in the form of your_program-report.json ' +
+    `in the dir ${__dirname}`);
+}
+
+function getTestStatus(test, pendingTitles) {
+  if(pendingTitles.includes(test.fullTitle)) {
+    return 'skipped';
+  }
+  const errorCount = Object.keys(test.err).length;
+  if(errorCount > 0) {
+    return 'failure';
+  }
+  return 'success';
+}
+
+// process each test file
+files.forEach(file => {
+  const implementation = file.match(/(.*)-report.json/)[1];
+  const results = JSON.parse(fs.readFileSync(
+    path.join(__dirname, file)), 'utf-8');
+  allResults[implementation] = {};
+  const pendingTitles = results.pending.map(t => t.fullTitle);
+
+  // process each test, noting the result
+  results.tests.forEach(test => {
+    allResults[implementation][test.fullTitle] =
+      getTestStatus(test, pendingTitles);
+    // assume vc.js tests all features
+    // TODO abstract this out
```

Comment:
if this TODO stays in, please be more specific so that it's possible for someone other than the author to know what it means.